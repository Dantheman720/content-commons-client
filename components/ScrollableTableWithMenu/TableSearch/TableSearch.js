import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  Button, Form, Grid, Input
} from 'semantic-ui-react';
import VisuallyHidden from 'components/VisuallyHidden/VisuallyHidden';
import './TableSearch.scss';

const TableSearch = props => {
  const [searchTerm, setSearchTerm] = useState( '' );

  const { handleSearchSubmit } = props;

  const handleChange = ( e, { value } ) => setSearchTerm( value );

  const handleSubmit = e => handleSearchSubmit( e, searchTerm );

  const handleClear = e => {
    setSearchTerm( '' );
    handleSearchSubmit( e, '' );
  };

  return (
    <Grid.Column floated="right" mobile={ 16 } computer={ 6 } className="tableSearch_wrapper">
      <Form onSubmit={ handleSubmit } size="tiny">
        <VisuallyHidden>
          <label htmlFor="search-projects">Search projects</label>
        </VisuallyHidden>
        <Input
          action
          className="tableSearch__searchTerm"
          fluid
          id="search-projects"
          onChange={ handleChange }
          placeholder="search term"
          value={ searchTerm }
        >
          <input />
          { searchTerm
            && (
              <Button
                aria-label="clear search"
                basic
                className="clear"
                type="button"
                icon="delete"
                onClick={ handleClear }
              />
            ) }
          <Button aria-label="search" icon="search" />
        </Input>
      </Form>
    </Grid.Column>
  );
};

TableSearch.propTypes = {
  handleSearchSubmit: PropTypes.func
};

export default TableSearch;
