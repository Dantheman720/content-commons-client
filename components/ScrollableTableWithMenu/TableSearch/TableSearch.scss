@import '../../../styles/colors.scss';

.ui.grid .row.items_tableSearch {
  padding-bottom: 0;
}

.tableSearch {
  &__searchTerm {
    > input {
      font-size: 12px;
      border-color: #5b616b !important;
    }

    > button.ui.button {
      padding: 12px !important;
      background-color: $light-blue;
      color: #fff;
      transition: color 0.4s;

      &.basic.clear {
        -webkit-box-shadow: 0px 0px 0px 1px #5b616b inset;
        box-shadow: 0px 0px 0px 1px #5b616b inset;
      }

      > i.icon {
        font-size: 20px;
      }

      &:hover {
        color: $blue;
      }
    }
  }
}