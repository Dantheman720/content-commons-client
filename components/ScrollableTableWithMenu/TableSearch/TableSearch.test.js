import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TableSearch from './TableSearch';

const props = { handleSearchSubmit: jest.fn() };

const Component = <TableSearch { ...props } />;

describe( '<TableSearch />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'submitting search Form calls handleSearchSubmit', () => {
    const wrapper = shallow( Component );
    const form = wrapper.find( 'Form' );

    form.simulate( 'submit' );
    expect( props.handleSearchSubmit ).toHaveBeenCalled();
  } );
} );
