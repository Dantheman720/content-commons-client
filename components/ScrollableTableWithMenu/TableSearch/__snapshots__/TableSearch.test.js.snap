// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableSearch /> renders without crashing 1`] = `
<GridColumn
  className="tableSearch_wrapper"
  computer={6}
  floated="right"
  mobile={16}
>
  <Form
    as="form"
    onSubmit={[Function]}
    size="tiny"
  >
    <VisuallyHidden
      el="div"
    >
      <label
        htmlFor="search-projects"
      >
        Search projects
      </label>
    </VisuallyHidden>
    <Input
      action={true}
      className="tableSearch__searchTerm"
      fluid={true}
      id="search-projects"
      onChange={[Function]}
      placeholder="search term"
      type="text"
      value=""
    >
      <input />
      <Button
        aria-label="search"
        as="button"
        icon="search"
      />
    </Input>
  </Form>
</GridColumn>
`;
